src/main/java/com/st/ims/utility/AppUtility.java

```diff
@@ -55,20 +61,60 @@ private static double getCommissionAmount(double invoiceAmount , double commissi
 		return invoiceAmount*(commissionPercentage/100);
 	}
 	
-	
-	public static InvoiceDetails setDefaultValues(InvoiceDetails invoiceDetail) {
+	/* This method will set default values for invoice object. */
+	public static Invoice setDefaultValues(Invoice invoice) {
 		try {
-			Invoice invoice = invoiceDetail.getInvoice();
 			invoice.setDueDate(getDueDate(invoice.getCreateDate()));
 			invoice.setCommissionAmount(getCommissionAmount(invoice.getInvoiceAmount(),invoice.getCommissionPercentage()));
 			invoice.setInvoiceNumber(getInvoiceNumber());
-			invoiceDetail.setInvoice(invoice);
-			return invoiceDetail;
+			return invoice;
 		} catch (NoSuchAlgorithmException e) {
-			e.printStackTrace();
+			logger.error("Exception while configuring dueDate,invoiceNumber and commissionAmount.",e);
 		}
-		return invoiceDetail;
-		
-		
+		return invoice;
+	}
+
+	/* This method will create a new InvoiceDetails List, so that we can persist invoice object with this list */
+	public static List<InvoiceDetails> getInvoiceDetailsList(Invoice invoice, List<InvoiceProducts> invoiceProductList){
+		List<InvoiceDetails> invoiceDetailsList = new ArrayList<>();
```

Comment:
why are u initiate List<InvoiceDetails> invoiceDetailsList = new ArrayList<>();

Reply:
It will be filled here with invoice,product and product counts and then it will be set to invoice and invoice will be persist in db.